torch/csrc/jit/tracer.cpp

```diff
@@ -61,27 +61,35 @@ struct TraceEval : autograd::Eval {
 
   virtual variable_list apply(const variable_list& inputs) override {
     auto should_trace = !flag.test_and_set();
-    if (should_trace) enterTrace(inputs);
-    auto outputs = Eval::apply(inputs);
-    if (should_trace) exitTrace(inputs, outputs);
+    if (!should_trace) {
+      return Eval::apply(inputs);
+    }
+    variable_list local_inputs = inputs;
+    enterTrace(local_inputs);
+    auto outputs = Eval::apply(local_inputs);
+    exitTrace(local_inputs, outputs);
     return outputs;
   }
 
-  void enterTrace(const variable_list& inputs) {
+  void enterTrace(variable_list& inputs) {
     auto tracing_state = weak_tracing_state.lock();
     if (!tracing_state) return;
 
     auto& graph = tracing_state->graph;
-    tracing_state->active = true;
     graph->advanceStage();
 
-    for (auto & input : inputs) {
+    for (std::size_t i = 0, num_inputs = inputs.size(); i < num_inputs; ++i) {
+      auto input = inputs[i];
```

Comment:
`auto& input`?

Reply:
I overwrite the input below. You can't overwrite references